Framework/PythonInterface/mantid/simpleapi.py

```diff
@@ -806,68 +850,72 @@ def ignore_property(name, ignore_regex):
         if _is_workspace_property(prop):
             value_str = prop.valueAsStr
             try:
-                retvals[name]=_api.AnalysisDataService[value_str]
+                retvals[name] = _api.AnalysisDataService[value_str]
             except KeyError:
                 if not prop.isOptional():
-                    raise RuntimeError("Internal error. Output workspace property '%s' on algorithm '%s' has not been stored correctly."
+                    raise RuntimeError("Internal error. Output workspace property '%s' on "
+                                       "algorithm '%s' has not been stored correctly."
                                        "Please contact development team." % (name,  algm_obj.name()))
         else:
             if hasattr(prop, 'value'):
-                retvals[name]=prop.value
+                retvals[name] = prop.value
             else:
-                raise RuntimeError('Internal error. Unknown property type encountered. "%s" on algorithm "%s" is not understood by '
-                       'Python. Please contact development team' % (name, algm_obj.name()))
-
-
+                raise RuntimeError('Internal error. Unknown property type encountered. "%s" '
+                                   'on algorithm "%s" is not understood by '
+                                   'Python. Please contact development team' % (name, algm_obj.name()))
     nvals = len(retvals)
     nlhs = lhs[0]
-    if nlhs > 1 and nvals != nlhs:
+
+    # If we have more than one value but not the same number of values throw
+    if 1 < nlhs != nvals:
         # There is a discrepancy in the number are unpacking variables
         # Let's not have the more cryptic unpacking error raised
         raise RuntimeError("%s is trying to return %d output(s) but you have provided %d variable(s). "
                            "These numbers must match." % (func_name, nvals, nlhs))
     if nvals > 0:
-        ret_type=namedtuple(func_name+"_returns",retvals.keys())
-        ret_value=ret_type(**retvals)
-        if nvals==1:
+        ret_type = namedtuple(func_name+"_returns", retvals.keys())
+        ret_value = ret_type(**retvals)
+        if nvals == 1:
```

Comment:
Also, I'm not sure I see the where the bug is? 

This checks if `nvals==1` which is if there is `1` value provided not if there is 1 value on the left hand side. I don't think the usage examples for [`GetEi`](http://docs.mantidproject.org/nightly/algorithms/GetEi-v2.html#usage) would work if this was broken.